apps/backend/src/application/application.service.ts

```diff
@@ -0,0 +1,139 @@
+import { BadRequestException, ForbiddenException, Injectable, NotFoundException, UseGuards } from '@nestjs/common';
+import { AuthGuard } from '@nestjs/passport';
+import { ApiBearerAuth } from '@nestjs/swagger';
+import { Application, Prisma, Role, User } from '@prisma/client';
+import { PrismaService } from 'nestjs-prisma';
+import { ApplicationPeriodService } from 'src/application-period/application-period.service';
+import { Roles } from 'src/auth/decorators/Roles.decorator';
+import { RolesGuard } from 'src/auth/roles.guard';
+
+import { CreateApplicationDto } from './dto/create-application.dto';
+import { UpdateApplicationDto } from './dto/update-application.dto';
+
+@Injectable()
+export class ApplicationService {
+  constructor(
+    private readonly prisma: PrismaService,
+    private readonly applicationPeriodService: ApplicationPeriodService
+  ) {}
+
+  async create(createApplicationDto: CreateApplicationDto, user: User) {
+    try {
+      const applicationPeriod = await this.prisma.applicationPeriod.findFirstOrThrow({
+        where: {
+          id: createApplicationDto.applicationPeriodId,
+        },
+      });
+      if (new Date(applicationPeriod.applicationPeriodEndAt) < new Date()) {
+        throw new BadRequestException('A jelentkezési időszak lejárt');
+      }
+      return await this.prisma.application.create({
+        data: {
+          user: {
+            connect: {
+              authSchId: user.authSchId,
+            },
+          },
+          applicationPeriod: {
+            connect: {
+              id: createApplicationDto.applicationPeriodId,
+            },
+          },
+        },
+      });
+    } catch (e) {
+      if (e instanceof Prisma.PrismaClientKnownRequestError) {
+        if (e.code === 'P2002') {
+          throw new BadRequestException('Ez a jelentkezés már létezik');
+        } else if (e.code === 'P2025') {
+          throw new NotFoundException('Nem található időszak');
+        }
+      }
+      if (e instanceof BadRequestException) {
+        throw e;
+      }
+      throw new BadRequestException('Nem sikerült létrehozni');
+    }
+  }
+
+  @UseGuards(AuthGuard('jwt'), RolesGuard)
+  @ApiBearerAuth()
+  @Roles(Role.BODY_ADMIN, Role.BODY_MEMBER)
+  async findAll() {
+    return await this.prisma.application.findMany();
+  }
+
+  async findOne(id: number): Promise<Application> {
+    const application = this.prisma.application.findFirst({
+      where: {
+        id,
+      },
+    });
```

Comment:
not sure about this, but it's possible that it's more efficient to use findUnique if you're querying by unique fields